src/hotspot/share/gc/shared/collectedHeap.cpp

```diff
@@ -409,30 +409,20 @@ void CollectedHeap::check_for_non_bad_heap_word_value(HeapWord* addr, size_t siz
 
 size_t CollectedHeap::max_tlab_size() const {
   // TLABs can't be bigger than we can fill with a int[Integer.MAX_VALUE].
-  // This restriction could be removed by enabling filling with multiple arrays.
-  // If we compute that the reasonable way as
-  //    header_size + ((sizeof(jint) * max_jint) / HeapWordSize)
-  // we'll overflow on the multiply, so we do the divide first.
-  // We actually lose a little by dividing first,
-  // but that just makes the TLAB  somewhat smaller than the biggest array,
-  // which is fine, since we'll be able to fill that.
-  size_t max_int_size = typeArrayOopDesc::header_size(T_INT) +
-              sizeof(jint) *
-              ((juint) max_jint / (size_t) HeapWordSize);
-  return align_down(max_int_size, MinObjAlignment);
-}
-
-size_t CollectedHeap::filler_array_hdr_size() {
-  return align_object_offset(arrayOopDesc::header_size(T_INT)); // align to Long
-}
-
-size_t CollectedHeap::filler_array_min_size() {
-  return align_object_size(filler_array_hdr_size()); // align to MinObjAlignment
+  return _filler_array_max_size;
 }
 
 void CollectedHeap::zap_filler_array_with(HeapWord* start, size_t words, juint value) {
-  Copy::fill_to_words(start + filler_array_hdr_size(),
-                      words - filler_array_hdr_size(), value);
+  int payload_start = arrayOopDesc::base_offset_in_bytes(T_INT);
+  if (!is_aligned(payload_start, BytesPerWord)) {
```

Comment:
So, this should also be `..., HeapWordSize` to match the post-correction assert, or do I miss some other intent?